docs/madr/decisions/018-fault-injection-policy.md

```diff
@@ -0,0 +1,230 @@
+#  Fault Injection policy
+
+- Status: accepted
+
+Technical Story: https://github.com/kumahq/kuma/issues/4734
+
+## Context and Problem Statement
+
+We want to create a [new policy matching compliant](https://github.com/kumahq/kuma/blob/22c157d4adac7f518b1b49939c7e9ea4d2a1876c/docs/madr/decisions/005-policy-matching.md)
+resource for managing fault injection.
+
+FaultInjection in Kuma is implemented with Envoy's [fault injection
+support](https://www.envoyproxy.io/docs/envoy/latest/configuration/http/http_filters/fault_filter).
+
+## Considered Options
+
+* Create a MeshFaultInjection
+
+## Decision Outcome
+
+Chosen option: create a MeshFaultInjection
+
+## Decision Drivers
+
+- Replace existing policies with new policy-matching compliant resources
+
+## Solution
+
+### Current configuration
+Below is a sample fault injection configuration.
+
+```yaml
+spec:
+  sources:
+    - match:
+        kuma.io/service: frontend_default_svc_80
+        kuma.io/protocol: http
+  destinations:
+    - match:
+        kuma.io/service: backend_default_svc_80
+        kuma.io/protocol: http
+  conf:
+    abort:
+      httpStatus: 500
+      percentage: 50
+    delay:
+      percentage: 50.5
+      value: 5s
+    responseBandwidth:
+      limit: 50 mbps
+      percentage: 50
+```
+
+The configuration translates to the Envoy configuration in Http's filters:
+
+```yaml
+{
+"name": "envoy.filters.http.fault",
+"typed_config": {
+  "@type": "type.googleapis.com/envoy.extensions.filters.http.fault.v3.HTTPFault",
+  "delay": {
+  "fixed_delay": "5s",
+  "percentage": {
+    "numerator": 505000,
+    "denominator": "TEN_THOUSAND"
+  }
+  },
+  "abort": {
+  "http_status": 500,
+  "percentage": {
+    "numerator": 50
+  }
+  },
+  "headers": [
+  {
+    "name": "x-kuma-tags",
+    "safe_regex_match": {
+    "google_re2": {},
+    "regex": ".*&kuma.io/protocol=[^&]*http[,&].*&kuma.io/service=[^&]*frontend_default_svc_80[,&].*"
+    }
+  }
+  ],
+  "response_rate_limit": {
+  "fixed_limit": {
+    "limit_kbps": "50000"
+  },
+  "percentage": {
+    "numerator": 50
+  }
+  }
+}
+```
+
+### Specification
+
+Fault injection supports only `HTTP` protocol and is configured in `HttpFilter`. We can use the header `x-kuma-tags` to match the specific source. In the case of `MeshHealthCheck` which doesn't set `x-kuma-tags` we can put `x-kuma-tags` or create `MeshFaultInjection` that allows only `Mesh` in `From` section.
```

Comment:
`MeshHealthCheck` supports only `targetRef.to`.

Reply:
@slonka It's more about https://github.com/kumahq/kuma/issues/5212

Reply:
@slonka ?

Reply:
The triage note says that it will be fixed in FaultInjection - https://github.com/kumahq/kuma/issues/5212#issuecomment-1297262410 . I did not do any specific implementation to fix this in [MeshHealthCheck](https://github.com/kumahq/kuma/pull/5503).

Reply:
So the underlying issue it relying on `x-kuma-tags` header to do the matching but if we match based on cert data then it's not an issue.

Reply:
But that assumes that mTLS is enabled always, which might not be true

Reply:
I think the header is simpler and we are already doing it for `FaultInjection` but without HealthCheck. Not sure how many changes requires getting data from the cert.

Reply:
> What do are we planning to do here? (cert data or header)

In terms of Fault Injection / HC compability we can:
1. Leave it as it is, just mention in the docs that HealthCheck/MeshHealthCheck does not work for fault injection at all
2. Base MeshFaultInjection on cert data and mention in the docs that `MeshHealthCheck` does not work without mtls.
3. Modify `MeshHealthCheck` so that it adds the header and it works for both mtls and non-mtls (it's in general ugly because we do some weird regex to match this but **it works™**).

I thought that we're going with 2, because of the comment linked above https://github.com/kumahq/kuma/issues/5212#issuecomment-1297262410 but I don't see the decision driver listed there so maybe @jakubdyszkiewicz can fill us in.

--- 

In terms of matching on specific service in targetRef policies:
1. I thought that we cannot right now match on specific service because that data is not available and that's why we need to get it from cert but I see that `MeshTrafficPermission` can match on [MeshService](https://github.com/kumahq/kuma/blob/db9de2b90fabf25c4e146d77120fc3889cb98612/pkg/plugins/policies/meshtrafficpermission/api/v1alpha1/validator.go#L39) in `From` so maybe I'm wrong. I think that @lobkovilya probably knows best here.

Reply:
MeshHealthCheck we are going to automatically set `x-kuma-tags` header.